controllers/storagecluster/cephblockpools.go

```diff
@@ -15,19 +15,51 @@ import (
 
 type ocsCephBlockPools struct{}
 
+// ensures that peer cluster secret is exist and adding into cephBlockPool
+func (r *StorageClusterReconciler) addPeerSecretsToCephBlockPool(initData *ocsv1.StorageCluster) (*cephv1.MirroringPeerSpec, error) {
+	secretNames := []string{}
+
+	if len(initData.Spec.Mirroring.SecretNames) == 0 {
+		err := fmt.Errorf("Mirroring peer cluster secretNames not found")
+		return nil, err
+	}
+	for _, secretName := range initData.Spec.Mirroring.SecretNames {
+		_, err := r.retrieveSecret(secretName, initData)
+		if err != nil {
+			return nil, err
```

Comment:
add more information about error. Something like, "peer cluster token could not be retrieved".